### PR TITLE
Issue #9512: update code example of AST for TokenTypes.METHOD_CALL

### PR DESCRIPTION
**Pull request by CryptoSingh1337**
Closes: #9512
image
```
**Test.java**
λ cat Test.java
public class Test {
        void test() {
                Integer.parseInt("123");
        }
}

**AST**
λ java -jar checkstyle-8.41-all.jar -t Test.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> Test [1:13]
`--OBJBLOCK -> OBJBLOCK [1:18]
    |--LCURLY -> { [1:18]
    |--METHOD_DEF -> METHOD_DEF [2:1]
    |   |--MODIFIERS -> MODIFIERS [2:1]
    |   |--TYPE -> TYPE [2:1]
    |   |   `--LITERAL_VOID -> void [2:1]
    |   |--IDENT -> test [2:6]
    |   |--LPAREN -> ( [2:10]
    |   |--PARAMETERS -> PARAMETERS [2:11]
    |   |--RPAREN -> ) [2:11]
    |   `--SLIST -> { [2:13]
    |       |--EXPR -> EXPR [3:18]
    |       |   `--METHOD_CALL -> ( [3:18]
    |       |       |--DOT -> . [3:9]
    |       |       |   |--IDENT -> Integer [3:2]
    |       |       |   `--IDENT -> parseInt [3:10]
    |       |       |--ELIST -> ELIST [3:19]
    |       |       |   `--EXPR -> EXPR [3:19]
    |       |       |       `--STRING_LITERAL -> "123" [3:19]
    |       |       `--RPAREN -> ) [3:24]
    |       |--SEMI -> ; [3:25]
    |       `--RCURLY -> } [4:1]
    `--RCURLY -> } [5:0]

# Printing the code that we care
λ java -jar checkstyle-8.41-all.jar -t Test.java | grep "3:"
    |       |--EXPR -> EXPR [3:18]
    |       |   `--METHOD_CALL -> ( [3:18]
    |       |       |--DOT -> . [3:9]
    |       |       |   |--IDENT -> Integer [3:2]
    |       |       |   `--IDENT -> parseInt [3:10]
    |       |       |--ELIST -> ELIST [3:19]
    |       |       |   `--EXPR -> EXPR [3:19]
    |       |       |       `--STRING_LITERAL -> "123" [3:19]
    |       |       `--RPAREN -> ) [3:24]
    |       |--SEMI -> ; [3:25]

```

### Expected update for javadoc
```
    |--EXPR -> EXPR
    |   `--METHOD_CALL -> (
    |       |--DOT -> .
    |       |   |--IDENT -> Integer
    |       |   `--IDENT -> parseInt
    |       |--ELIST -> ELIST
    |       |   `--EXPR -> EXPR
    |       |       `--STRING_LITERAL -> "123"
    |       `--RPAREN -> )
    |--SEMI -> ;
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -691,19 +691,27 @@
      *
      * <p>For example:</p>
      * <pre>
-     * Math.random()
+     * String n = Integer.parseInt("123");
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:694]: The example should be as short as possible. Please remove variable assignment.

**Review by pbludov — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (CryptoSingh1337) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:694]: Can I take Math.random() only.

Comment (CryptoSingh1337) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:694]: Is this correct?
```
**Test.java**
public class Test {
	void test() {
		Integer.parseInt("123");
	}
}

# Expected update
 `--METHOD_CALL -> (
     |--DOT -> .
     |   |--IDENT -> Integer
     |   `--IDENT -> parseInt
     |--ELIST -> ELIST
     |   `--EXPR -> EXPR
     |       `--STRING_LITERAL -> "123"
     `--RPAREN -> )
```

**Comment by theanmolsharma** [comment]
hey, I am trying to fix #9549 . CI tests are not passing. Could you please have a look? Also, Which IDE do you use? I use IntelliJ Ultimate.

**Comment by CryptoSingh1337** [comment]
image
```
**Test.java**
λ cat Test.java
public class Test {
        void test() {
                Integer.parseInt("123");
        }
}

**AST**
λ java -jar checkstyle-8.41-all.jar -t Test.java
CLASS_DEF -> CLASS_DEF [1:0]
|--MODIFIERS -> MODIFIERS [1:0]
|   `--LITERAL_PUBLIC -> public [1:0]
|--LITERAL_CLASS -> class [1:7]
|--IDENT -> Test [1:13]
`--OBJBLOCK -> OBJBLOCK [1:18]
    |--LCURLY -> { [1:18]
    |--METHOD_DEF -> METHOD_DEF [2:1]
    |   |--MODIFIERS -> MODIFIERS [2:1]
    |   |--TYPE -> TYPE [2:1]
    |   |   `--LITERAL_VOID -> void [2:1]
    |   |--IDENT -> test [2:6]
    |   |--LPAREN -> ( [2:10]
    |   |--PARAMETERS -> PARAMETERS [2:11]
    |   |--RPAREN -> ) [2:11]
    |   `--SLIST -> { [2:13]
    |       |--EXPR -> EXPR [3:18]
    |       |   `--METHOD_CALL -> ( [3:18]
    |       |       |--DOT -> . [3:9]
    |       |       |   |--IDENT -> Integer [3:2]
    |       |       |   `--IDENT -> parseInt [3:10]
    |       |       |--ELIST -> ELIST [3:19]
    |       |       |   `--EXPR -> EXPR [3:19]
    |       |       |       `--STRING_LITERAL -> "123" [3:19]
    |       |       `--RPAREN -> ) [3:24]
    |       |--SEMI -> ; [3:25]
    |       `--RCURLY -> } [4:1]
    `--RCURLY -> } [5:0]

# Printing the code that we care
λ java -jar checkstyle-8.41-all.jar -t Test.java | grep "3:"
    |       |--EXPR -> EXPR [3:18]
    |       |   `--METHOD_CALL -> ( [3:18]
    |       |       |--DOT -> . [3:9]
    |       |       |   |--IDENT -> Integer [3:2]
    |       |       |   `--IDENT -> parseInt [3:10]
    |       |       |--ELIST -> ELIST [3:19]
    |       |       |   `--EXPR -> EXPR [3:19]
    |       |       |       `--STRING_LITERAL -> "123" [3:19]
    |       |       `--RPAREN -> ) [3:24]
    |       |--SEMI -> ; [3:25]

```

### Expected update for javadoc
```
    |--EXPR -> EXPR [3:18]
    |   `--METHOD_CALL -> ( [3:18]
    |       |--DOT -> . [3:9]
    |       |   |--IDENT -> Integer [3:2]
    |       |   `--IDENT -> parseInt [3:10]
    |       |--ELIST -> ELIST [3:19]
    |       |   `--EXPR -> EXPR [3:19]
    |       |       `--STRING_LITERAL -> "123" [3:19]
    |       `--RPAREN -> ) [3:24]
    |--SEMI -> ; [3:25]
```

**Comment by CryptoSingh1337** [comment]
> hey, I am trying to fix #9549 . CI tests are not passing. Could you please have a look? Also, Which IDE do you use? I use IntelliJ Ultimate.

I am also unable to find any error. It is looking fine.
I am also using Intellij IDEA Ultimate.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 1 above)
Comment (pbludov) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java]: yes, this looks very clear.

**Review by romani — APPROVED** [review]

**Pull request merged**